server/odc-core/src/main/java/com/oceanbase/odc/core/datasource/SingleConnectionDataSource.java

```diff
@@ -235,6 +261,38 @@ private synchronized Connection innerCreateConnection() throws SQLException {
         }
     }
 
+
+    private void initKeepAliveScheduler() {
+        if (!keepAlive) {
+            return;
+        }
+        failedKeepAliveAttempts.set(0);
```

Comment:
this method seems not be re-entried?
will the `keepAliveScheduler` be shutdown when I try to call this method twice?

Reply:
Will not reconstruct the scheduler when reset connection.